base/special/exp.jl

```diff
@@ -114,13 +118,13 @@ function exp{T<:Union{Float32,Float64}}(x::T)
         # scale back
         if k > -significand_bits(T)
             # multiply by 2.0 first to prevent overflow, which helps extends the range
-            k == exponent_max(T) && return y*T(2.0)*T(2.0)^(exponent_max(T) - 1)
+            k == exponent_max(T) && return y * T(2.0) * T(2.0)^(exponent_max(T) - 1)
             twopk = reinterpret(T, rem(exponent_bias(T) + k, fpinttype(T)) << significand_bits(T))
             return y*twopk
         else
             # add significand_bits(T) + 1 to lift the range outside the subnormals
             twopk = reinterpret(T, rem(exponent_bias(T) + significand_bits(T) + 1 + k, fpinttype(T)) << significand_bits(T))
-            return y*twopk*T(2.0)^(-significand_bits(T) - 1)
+            return y * twopk * T(2.0)^(-significand_bits(T) - 1)
```

Comment:
> @musm I'm still not sure I quite understand the issue here. All these expressions are still being constant folded. e.g. on this branch:

I'm a little confused then why we need:
https://github.com/JuliaLang/julia/pull/19890/files#diff-4da275bf96d6e97c1918f46754df04d1R68
in the `exp` function since 
https://github.com/JuliaLang/julia/blob/1282205a0285ffacd431f5e0a13cbcf8524468f2/base/special/exp.jl#L121
and 
https://github.com/JuliaLang/julia/blob/1282205a0285ffacd431f5e0a13cbcf8524468f2/base/special/exp.jl#L127
should also be constant folded (i.e. `T(2.0)^(exponent_max(T) - 1)` and  `T(2.0)^(-significand_bits(T) - 1)` ). So doesn't that make the`    ^ = Base.FastMath.pow_fast` unnecessary  (the benchmarks showed otherwise...?)  (here's the def for `exponent_max` https://github.com/JuliaLang/julia/blob/1282205a0285ffacd431f5e0a13cbcf8524468f2/base/float.jl#L751)

@tkelman 
> that commit seems like an improvement in readability to me
The reason I suggested the current code is more clear is that it's really obvious what's currently happening with the `signficand_bits` and the scaling, e.g.  see the line and the following one
https://github.com/JuliaLang/julia/blob/master/base/math.jl#L556
it may be less readable, but I believe it's more clear and would be  a lot easier to quickly infer what it's doing upon first glance or in the future when revisiting the logic. 

This type of scaling is also used in almost the exact same logic, repeatedly, in the `exp` function https://github.com/JuliaLang/julia/blob/1282205a0285ffacd431f5e0a13cbcf8524468f2/base/special/exp.jl#L118
I'd prefer to keep as it as is currently, if possible, due to these reasons.